### DISCUSSION TITLE
Cancelling mutations in progress on unsubscribe.

### DISCUSSION THREAD

**Discussion by jzarzeckis** [urql-graphql/urql#1500] [Q&A]
I haven't looked that much into `urql` source code - but in devtools I see that requests for Queries that are unsubscribed while in progress do get cancelled.

I was hoping this would also be the case for mutations - but as far as I see - even if I unsubscribe to these mutation results - the http requests remain pending. I suppose this is intentional, as I may want to save a form, and navigate away to a different section while sending, and still expect the form content to be sent? Is there any way to enable mutation cancellation on unsubscribe of observable?

**Comment by jzarzeckis** [comment]
Or am I supposed to create a wrapper that will issue a [teardown](https://formidable.com/open-source/urql/docs/api/core/#operationtype) operation on unsubscribe on my own?

**Comment by jonathanstanley** [comment]
i may have the same issue.  I have created a mutation that unsubscribes when the response is received.  once unsubscribed, the mutation ceases to update (as expected).

however, in devtools, it appears that the mutation does not unusbscribe:

https://user-images.githubusercontent.com/2755958/160538485-00db0c91-f8b9-409d-b207-ac0d24d8d508.mov

i can't tell if this is something wrong in wonka, urql, or urql-devtools.  when i do the same process with query, devtools shows the query is torn down. it seems it is just mutation with the issue.
